src/main/pages/che-7/end-user-guide/proc_mount-secret-as-a-file-in-workspace-container.adoc

```diff
@@ -0,0 +1,86 @@
+[id="mount-secret-as-file-in-workspace-container_{context}"]
+= Mount secret as file into workspace container
+
+This section describes how to mount an secret from user-s namespace as a file in a single or
+multiple workspace containers of {prod}.
+
+.Prerequisites
+
+* A running instance of {prod}. To install an instance of {prod}, see link:{site-baseurl}che-7/che-quick-starts/[{prod-short} 'quick-starts'].
+
+.Procedure
+
+* Create secret in user namespace following those rules :
+
+Labels must match the set of labels configured in `che.workspace.provision.secret.labels`
+property of {prod}. By default it is a set of two labels:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mvn-settings-secret
+  labels:
+    app.kubernetes.io/part-of: che.eclipse.org
+    app.kubernetes.io/component: workspace-secret
+...
+----
+
+
+Annotations must contain specific items, indicating that given secret should be mount as file,
+providing mount path, and, optionally, specify container name where this mount should be applied.
+If there is no target-container annotation, secret will be mount into all user containers of the workspace.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mvn-settings-secret
+  annotations:
+    che.eclipse.org/target-container: maven
+    che.eclipse.org/mount-path: /home/user/.m2/
+    che.eclipse.org/mount-as: file
+  labels:
+...
+----
+
+Secret data must have exactly one item, and its name must match desired filename mount into container:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mvn-settings-secret
+  annotations:
+    ...
+  labels:
+    ...
+data:
+  settings.xml: | ...<base64 encoded data content here>...
+----
+
+So, complete example how the secret must looks like:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mvn-settings-secret
+  labels:
+    app.kubernetes.io/part-of: che.eclipse.org
+    app.kubernetes.io/component: workspace-secret
+  annotations:
+    che.eclipse.org/target-container: maven
+    che.eclipse.org/mount-path: /home/user/.m2/
+    che.eclipse.org/mount-as: file
+data:
+  settings.xml: | ...<base64 encoded data content here>...
+----
+
+
+This will resulting in file named `settings.xml` will be mount at path `/home/user/.m2/` of the
+user container named `maven`.
```

Comment:
Should we mention about the folder permission that would be after mounting this file?

Reply:
would be addressed as part of this issue https://github.com/eclipse/che/issues/16857